patches/tModLoader/Terraria/GameContent/UI/Elements/UIAchievementListItem.cs.patch

```diff
@@ -0,0 +1,123 @@
++			nameText = "???";
++			descriptionText = "???";
++		}
++
+-		string text = FontAssets.ItemStack.Value.CreateWrappedText(_achievement.Description.Value, (num3 - 20f) * (1f / baseScale2.X), Language.ActiveCulture.CultureInfo);
++		string text = FontAssets.ItemStack.Value.CreateWrappedText(descriptionText, (num3 - 20f) * (1f / baseScale2.X), Language.ActiveCulture.CultureInfo);
+ 		Vector2 stringSize = ChatManager.GetStringSize(FontAssets.ItemStack.Value, text, baseScale2, num3);
+ 		if (!_large)
+-			stringSize = ChatManager.GetStringSize(FontAssets.ItemStack.Value, _achievement.Description.Value, baseScale2, num3);
++			stringSize = ChatManager.GetStringSize(FontAssets.ItemStack.Value, descriptionText, baseScale2, num3);
+ 
+ 		float num4 = 38f + (float)(_large ? 20 : 0);
+ 		if (stringSize.Y > num4)
+@@ -104,7 +_,11 @@
+ 		spriteBatch.Draw(_categoryTexture.Value, vector3, _categoryTexture.Frame(4, 2, (int)category), base.IsMouseHovering ? Color.White : Color.Silver, 0f, Vector2.Zero, 0.5f, SpriteEffects.None, 0f);
+ 		vector3.X += 4f;
+ 		vector3.X += 17f;
+-		ChatManager.DrawColorCodedStringWithShadow(spriteBatch, FontAssets.ItemStack.Value, _achievement.FriendlyName.Value, vector3, value, 0f, Vector2.Zero, baseScale, num3);
++		ChatManager.DrawColorCodedStringWithShadow(spriteBatch, FontAssets.ItemStack.Value, nameText, vector3, value, 0f, Vector2.Zero, baseScale, num3);
++		string ModName = _achievement.Modded ? _achievement.ModAchievement.Mod.DisplayNameClean : "Terraria";
++		_offsetProgressBarX = ((ModName.Length + 1) * 10);
++		Vector2 ModNameLocation = new Vector2(vector3.X + num3 - _offsetProgressBarX, vector3.Y);
++		ChatManager.DrawColorCodedStringWithShadow(spriteBatch, FontAssets.ItemStack.Value, ModName, ModNameLocation, value, 0f, Vector2.Zero, baseScale, num3);
+ 		vector3.X -= 17f;
+ 		Vector2 position = vector2 + Vector2.UnitY * 27f + vector;
+ 		DrawPanelBottom(spriteBatch, position, num3, color);
+@@ -126,14 +_,15 @@
+ 			if (!base.IsMouseHovering)
+ 				color3 = Color.Lerp(color3, Color.Black, 0.25f);
+ 
+-			DrawProgressBar(spriteBatch, progress, vector4 - Vector2.UnitX * num5 * 0.7f, num5, color3, color2, color2.MultiplyRGBA(new Color(new Vector4(1f, 1f, 1f, 0.5f))));
++			DrawProgressBar(spriteBatch, progress, vector4 - (Vector2.UnitX * num5 * 0.7f), num5, color3, color2, color2.MultiplyRGBA(new Color(new Vector4(1f, 1f, 1f, 0.5f))));
+-			vector4.X -= num5 * 1.4f + stringSize2.X;
++			vector4.X -= num5 * 1.4f + stringSize2.X + _offsetProgressBarX - 4f;
+ 			ChatManager.DrawColorCodedStringWithShadow(spriteBatch, FontAssets.ItemStack.Value, text2, vector4, value, 0f, new Vector2(0f, 0f), baseScale3, 90f);
+ 		}
+ 	}
+ 
+ 	private void UpdateIconFrame()
+ 	{
++		if (_achievement.Modded) return;
+ 		if (!_locked)
+ 			_iconFrame = _iconFrameUnlocked;
+ 		else
+@@ -202,6 +_,7 @@
+ 
+ 		if (BackColor == Color.Transparent)
+ 			FillingColor = new Color(255, 255, 255);
++		spot.X -= _offsetProgressBarX - 16;
```

Comment:
Recommend dropping in a new line and doing `// Added by TML` for little surprise things like this where it's not instantly obvious. Also code convention is a blank line after reducing indents